### ISSUE TITLE
Add cache to Ghost blog service

### ISSUE THREAD

**Issue by BretFisher** [BretFisher/dogvscat#8]
Since the Ghost service can only have a single replica (Ghost design limitation), it's expected to have a caching proxy in front of it, which itself could be multi-replica for better LB/HA. I'm thinking the best options here are varnish or nginx, which would be another service in the Ghost stack file.

**Comment by pascalandy** [comment]
I'll be very curious to see the architecture you have in mind. There is [an issue](https://github.com/containous/traefik/issues/878#issuecomment-276785455) on Traefik's GitHub but it goes nowhere at this point.

**Comment by BretFisher** [comment]
I was assuming a cache container in front of Ghost node container. So traffic would be:

client --> cloud LB-->swarm ingress-->traefik-->proxy overlay-->cache-->ghost overlay-->ghost

I'm normally a nginx guy, and have done reverse proxy caching with it before, but thought it would be fun to try varnish in this project, which I haven't used.

Hopefully, we can use one of these images:

* [This one](https://github.com/coopTilleuls/docker-varnish) requires a varnish config file added as a swarm config, but looks flexible and up to date.

* [This one is ENVVAR heavy](https://github.com/wodby/varnish), which I like (very 12factor.net!)  No configs needed hopefully.

**Comment by BretFisher** [comment]
Yes, it would be a 1-to-1. The other option would be to replace layer 7 reverse proxy (Traefik) with something that can do both DNS routing and caching... like nginx... but I feel like that's harder to get right.

**Comment by pascalandy** [comment]
Thanks for your input on this :)
BTW, I'll will not replace Traefik anytime soon :-p

**Comment by pascalandy** [comment]
I'm not very knowledgeable with Caddy, but it sounds like an elegant solution
https://github.com/caddyserver/examples/tree/master/ghost
